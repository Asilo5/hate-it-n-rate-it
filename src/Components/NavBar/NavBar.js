import React from 'react';
import { NavLink } from 'react-router-dom';
import './NavBar.scss'

const NavBar = () => {
  return (
    <div className='NavBar'>
      <NavLink className='nav_button' to='/login'> Sign In </NavLink>
    </div>
  )
}

export default NavBar;