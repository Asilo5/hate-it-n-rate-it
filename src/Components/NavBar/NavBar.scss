.NavBar {
  
  .nav_button {
    background-color: #5d5c5dab;
    border-radius: 10px;
    color: white;
    font-size: 25px;
    height: 30px;
    padding: 7px 15px;
    text-decoration: none;
    width: 150px;
  }
}
.NavBar:hover {
    transform: scale(1.05);
}