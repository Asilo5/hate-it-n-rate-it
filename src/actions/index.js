export const addMoviesData = movies => ({
  type: 'ADD_MOVIES_DATA',
  movies
})

export const setUser = user => ({
  type: 'SET_USER',
  user
})

export const hasError = (error) => ({
  type: 'HAS_ERROR',
  error
})

