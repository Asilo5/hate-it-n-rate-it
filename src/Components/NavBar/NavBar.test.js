import React from 'react';
import { shallow } from 'enzyme';
import { NavBar, mapDispatchToProps } from './NavBar';
import { setUser } from '../../actions/index';


describe('NavBar', () => {
  let wrapper;
  let mockUser = {};
  const mockSetUser = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<NavBar user={mockUser} setUser={mockSetUser} />);
  });

  describe('snapshots', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapDispatchToProps', () => {
   it('calls dispatch with a setUser action', () => {
    mockUser = {
      name: 'John Adams',
      email: 'theREALJohnA@gmail,com',
      password: 'presidentsarechumps'
    }
    const mockDispatch = jest.fn();
    const actionToDispatch = setUser(mockUser);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.setUser(mockUser);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
   })

  })
})