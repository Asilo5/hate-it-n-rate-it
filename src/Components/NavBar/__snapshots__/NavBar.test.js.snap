// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar snapshots should match the snapshot 1`] = `
<div
  className="NavBar"
>
  <NavLink
    className="nav_button"
    to="/login"
  >
     Sign In 
  </NavLink>
  <NavLink
    className="home_button"
    to="/"
  >
     Home 
  </NavLink>
</div>
`;
